### PR TITLE
[automotive] align to skill template

### PR DESCRIPTION
**Pull request by xhr0804**
### Purpose
*What is the context of this pull request? Why is it being done?*

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: skills/csharp/experimental/automotiveskill/AutomotiveSkill.csproj [hunk 1]
```diff
@@ -14,23 +14,11 @@
...
-    <PackageReference Include="Microsoft.Bot.Builder.Dialogs" Version="4.8.1" />
     <PackageReference Include="Microsoft.Bot.Builder.Integration.ApplicationInsights.Core" Version="4.8.1" />
-    <PackageReference Include="Microsoft.Bot.Builder.Integration.AspNet.Core" Version="4.8.1" />
-    <PackageReference Include="Microsoft.Bot.Builder.LanguageGeneration" Version="4.8.1-rc0" />
-    <PackageReference Include="Microsoft.Bot.Configuration" Version="4.8.1" />
-    <PackageReference Include="Microsoft.Bot.Connector" Version="4.8.1" />
-    <PackageReference Include="Microsoft.Bot.Schema" Version="4.8.1" />
     <PackageReference Include="Microsoft.Bot.Solutions" Version="0.9.0-daily252" />
-    <PackageReference Include="Microsoft.Graph" Version="1.15.0" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
```
Comment (bobokids) [inline skills/csharp/experimental/automotiveskill/AutomotiveSkill.csproj]: add    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118" PrivateAssets="all" />

File: skills/csharp/experimental/automotiveskill/Adapters/DefaultAdapter.cs [hunk 2]
```diff
@@ -1,44 +1,124 @@
...
+        }
+
+        private async Task SendErrorMessageAsync(ITurnContext turnContext, Exception exception)
+        {
+            try
+            {
+                _telemetryClient.TrackException(exception);
+
+                // Send a message to the user.
+                await turnContext.SendActivityAsync(_templateEngine.GenerateActivityForLocale("ErrorMessage"));
```
Comment (bobokids) [inline skills/csharp/experimental/automotiveskill/Adapters/DefaultAdapter.cs]: >"ErrorMessage" [](start = 94, length = 14)

AutomotiveSkillSharedResponses.ErrorMessage

**Comment by bobokids** [comment]
>         private IStatePropertyAccessor<DialogState> _dialogStateAccessor;

readonly

---
Refers to: skills/csharp/experimental/automotiveskill/Bots/DefaultActivityHandler.cs:22 in 8842cac. [](commit_id = 8842cac5f9e291c2fa78666c1a6c287f2fbb359f, deletion_comment = False)

File: skills/csharp/experimental/automotiveskill/Dialogs/AutomotiveSkillDialogBase.cs [hunk 3]
```diff
@@ -8,31 +8,30 @@
...
+            Services = serviceProvider.GetService<BotServices>();
+            TemplateManager = serviceProvider.GetService<LocaleTemplateManager>();
+
+            // Initialize skill state
+            var conversationState = serviceProvider.GetService<ConversationState>();
             Accessor = conversationState.CreateProperty<AutomotiveSkillState>(nameof(AutomotiveSkillState));
-            TelemetryClient = telemetryClient;
         }
 
         protected BotSettings Settings { get; set; }
```
Comment (bobokids) [inline skills/csharp/experimental/automotiveskill/Dialogs/AutomotiveSkillDialogBase.cs]: >t [](start = 48, length = 1)

only keep { get; }

**Comment by bobokids** [comment]
>         private class Events

static

---
Refers to: skills/csharp/experimental/automotiveskill/Dialogs/MainDialog.cs:321 in 8842cac. [](commit_id = 8842cac5f9e291c2fa78666c1a6c287f2fbb359f, deletion_comment = False)

File: skills/csharp/experimental/automotiveskill/Dialogs/VehicleSettingsDialog.cs [hunk 4]
```diff
@@ -323,9 +315,9 @@ private async Task<DialogTurnResult> ProcessVehicleSettingsChange(WaterfallStepC
...
         /// <param name="cancellationToken">Cancellation Token.</param>
         /// <returns>Whether prompt value was validated.</returns>
-        private async Task<bool> SettingValueSelectionValidator(PromptValidatorContext<FoundChoice> promptContext, CancellationToken cancellationToken)
+        private async Task<bool> SettingValueSelectionValidatorAsync(PromptValidatorContext<FoundChoice> promptContext, CancellationToken cancellationToken)
         {
-            var state = await Accessor.GetAsync(promptContext.Context);
+            var state = await Accessor.GetAsync(promptContext.Context, cancellationToken: cancellationToken);
 
             // Use the value selection LUIS model to perform validation of the users entered setting value
             var valueSelectionResult = await vehicleSettingValueSelectionLuisRecognizer.RecognizeAsync<SettingsValueLuis>(promptContext.Context, CancellationToken.None);
```
Comment (bobokids) [inline skills/csharp/experimental/automotiveskill/Dialogs/VehicleSettingsDialog.cs]: using cancellationToken

File: skills/csharp/experimental/automotiveskill/AutomotiveSkill.csproj (hunk 1 above)
Comment (xhr0804) [reply skills/csharp/experimental/automotiveskill/AutomotiveSkill.csproj]: fixed

File: skills/csharp/experimental/automotiveskill/AutomotiveSkill.csproj [hunk 5]
```diff
@@ -14,25 +14,14 @@
...
     <PackageReference Include="SharpYaml" Version="1.6.5" />
+    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118" PrivateAssets="all" />
```
Comment (lzc850612) [inline skills/csharp/experimental/automotiveskill/AutomotiveSkill.csproj]: can we move this to the Directory.Build.Props file so we can remove this from all skills?

Comment (bobokids) [reply skills/csharp/experimental/automotiveskill/AutomotiveSkill.csproj]: Good point. we may open another issue to track moving all projects shared properties to this file.

**Pull request merged**